lib/network/modules/EdgesHandler.js

```diff
@@ -367,6 +367,16 @@ class EdgesHandler {
   }
 
   create(properties) {
+    // It is not at all clear why all these separate options should be passed:
+    //
+    // - this.edgeOptions is set in setOptions()
+    //   the value of which is also added to this.options with parseOptions() 
+    // - this.defaultOptions has been added to this.options with util.extend() in ctor
+    //
+    // So, in theory, this.options should be enough.
+    // The only reason I can think of for this, is that precedence is important.
+    // TODO: make unit tests for this, to check if edgeOptions and defaultOptions are redundant
+    //
     return new Edge(properties, this.body, this.options, this.defaultOptions, this.edgeOptions)
```

Comment:
OK. This is something I intend to pick up in due time, the comment serves as a reminder/warning. It's just that it was IMHO outside of the scope of the issue, and I was doing too much already.